ext/uri/php_uri.c

```diff
@@ -1011,14 +1015,14 @@ ZEND_ATTRIBUTE_NONNULL PHPAPI void php_uri_implementation_set_object_handlers(ze
 	object_handlers->clone_obj = uri_clone_obj_handler;
 }
 
-zend_result uri_handler_register(const uri_handler_t *uri_handler)
+PHPAPI zend_result uri_handler_register(const uri_handler_t *uri_handler)
 {
-	zend_string *key = zend_string_init_interned(uri_handler->name, strlen(uri_handler->name), 1);
+	zend_string *key = zend_string_init_interned(uri_handler->name, strlen(uri_handler->name), true);
 
-	ZEND_ASSERT(uri_handler->name != NULL);
+	ZEND_ASSERT(uri_handler->name != NULL && (strlen(uri_handler->name) > 0 || strcmp(uri_handler->name, URI_PARSER_PHP) == 0));
 	ZEND_ASSERT(uri_handler->parse_uri != NULL);
-	ZEND_ASSERT(uri_handler->clone_uri != NULL);
-	ZEND_ASSERT(uri_handler->uri_to_string != NULL);
+	ZEND_ASSERT(uri_handler->clone_uri != NULL || strcmp(uri_handler->name, URI_PARSER_PHP) == 0);
```

Comment:
I wanted to keep these assertions even though two handlers of the legacy parser became NULL. So I excluded this URI handler from the checks.

Reply:
Yes, I can understand your POV. Do you think it's better to get rid of the latter two special cased assertions?